### ISSUE TITLE
webdav plugin limits

### ISSUE THREAD

**Issue by angrybb** [syncany/syncany#361]
So in talk on irc i mentioned what are the limits of my ISP(German 1&1) webdav and @cr0 said there was also problem with dropbox max nr. of files limit and that i should open issue for this. So here it is.
1&1 Online Speicher webdav limits can be found here (in german): http://bit.ly/18yyIJN
- max 300 000 of total files and folders combined
- max 3 000 files and subfolders inside one folder
- max single file size 2GB
- max 255 chars long file/folder name
- not allowed chars in filename: ? > |* > : " \ / +

As those limits may be different on other webdav server it would also be nice if those could be set manualy by user (on cmd line or in gui)

**Comment by cr0** [comment]
relates to #353

**Comment by cr0** [comment]
Should we make *folderization*  optional or required? More precisely, do we need a boolean option?

**Comment by pimotte** [comment]
If we do this properly, we can include it in -lib using patterns similarly to the RetriableTM and TransactionAwareTM. In this case the limit become an option of every plugin. We can then assume these limits are configured correctly, and then we can do the computation wheter folderization is necessary based on the limits and some educated guesses on maximum repo size and average multichunk size (basically what Philipp did in the excel sheet). We could even have these as parameters, such that a user can say "I want to be able to store 100GB in this repo"

**Comment by cr0** [comment]
Yep, it feels much more natural to put this stuff into the core!

Am I right that [this](https://github.com/syncany/syncany/blob/develop/syncany-lib/src/main/java/org/syncany/operations/AbstractTransferOperation.java#L84) is the only place where transfer managers are instantiated? If so we could allow `TransferManager`s  to implement a `Folderizable` interface which instructs the `AbstractTransferOperation` to wrap the `TransactionAwareTransferManager` in a `FolderizableTransferManager` again. However, `list` operations will be very hard to conduct but we'll also find a solution for that.
`Folderizable` could also force the `<Plugin>TransferManager` to provide values for folder depth, bytes per folder and so on. If a plugin dev wants to take some user inputs into account to derive these values is his decision.

**Comment by pimotte** [comment]
Afaik, yes that is the only place we instatiate those (there and in the test suite).

**Comment by cr0** [comment]
I started on this in https://github.com/syncany/syncany/commit/5b0ebd4e87a14fa215e4a79f9f533fa94ec1a4c7

Untested but I need to catch my beer now :D

**Comment by cr0** [comment]
**on hold** until #366 is implemented.

**Comment by cr0** [comment]
Now we can continue with this issue because we've finally merged #452
